account_balance_reporting/models/account_balance_reporting_report.py

```diff
@@ -206,19 +206,19 @@ def name_get(self):
             res.append((item.id, "[%s] %s" % (item.code, item.name)))
         return res
 
-    def name_search(self, cr, uid, name, args=None, operator='ilike',
-                    context=None, limit=80):
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
         """Redefine the method to allow searching by code."""
-        ids = []
         if args is None:
             args = []
-        if name:
-            ids = self.search(cr, uid, [('code', 'ilike', name)] + args,
-                              limit=limit, context=context)
-        if not ids:
-            ids = self.search(cr, uid, [('name', operator, name)] + args,
-                              limit=limit, context=context)
-        return self.name_get(cr, uid, ids, context=context)
+        return self.search(
+            [
+                '|',
+                ('code', operator, name),
+                ('name', operator, name),
+            ] + args,
+            limit=limit
+        ).name_get()
```

Comment:
Porque en args sólo vienen los AND extra, pero la búsqueda principal por el campo name no va ahí, y por tanto, no se puede hacer medio OR.